### PR TITLE
feat: add new params to completion snippet

### PR DESCRIPTION
**Pull request by p-spacek**
### What does this PR do?
When object is type anyOf (array: null | string[]) some code completion doesn't work properly.
For example snippets on array it's not working very well.
```yaml
"anyOf_arrayObj": {
  "anyOf": [
    {
      "type": "array",
      "items": {
        "type": "object"
      }
    },
    {
      "type": "null"
    }
  ],
  "defaultSnippets": [
    {
      "label": "- (array item)",
      "type": "array",
      "suggestionKind": 9,
      "body": {
        "key": ""
      }
    }
  ]
},
```
Helps also in scenarios where yaml parser doesn't know that parent is array:
```yaml
anyOf_arrayObj:
  # invoke array completion here
```

This new parameters can specify:
 - `type`: type of snippet.
 - `symbolKind`: id of symbol that is displayed on completion panel. 
 
This feature probably doesn't solve the cause of the problem. But offers workaround.

### What issues does this PR fix or reference?
[Completion-Snippet: add parameters to defaultSnippets to specify the type and simbolKind](https://github.com/p-spacek/yaml-language-server/issues/7)

### Is it tested? How?
added unit test

### CODE REVIEW

File: src/languageservice/jsonSchema.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import { CompletionItemKind } from 'vscode-json-languageservice';
```
Comment (evidolob) [inline src/languageservice/jsonSchema.ts:1]: Could you move import under license header

Comment (p-spacek) [reply src/languageservice/jsonSchema.ts]: sorry, done

**Review by p-spacek — COMMENTED** [review]
done

File: src/languageservice/services/yamlCompletion.ts [hunk 2]
```diff
@@ -535,7 +535,7 @@ export class YAMLCompletion extends JSONCompletion {
           filterText = insertText.replace(/[\n]/g, ''); // remove new lines
         }
         collector.add({
-          kind: this.getSuggestionKind(type),
+          kind: s.suggestionKind || this.getSuggestionKind(type),
```
Comment (JPinkney) [inline src/languageservice/services/yamlCompletion.ts:538]: Is this change needed for this PR? If so I think we should have a simple test case covering it just to make sure everything works

Comment (p-spacek) [reply src/languageservice/services/yamlCompletion.ts:538]: I'm using it with custom array snippets where I need to set custom symbol.
I added UT for this case.

If you think that custom suggestionKind shouldn't be there. I can remove it and add it locally.

**Review by JPinkney — APPROVED** [review]

File: src/languageservice/services/yamlCompletion.ts (hunk 2 above)
Comment (JPinkney) [reply src/languageservice/services/yamlCompletion.ts:538]: I think you can keep it in

**Comment by evidolob** [comment]
@p-spacek Sorry for the delay, there are some tests fail. Could you fix them ASAP? I want to include this PR in to next release.

**Comment by p-spacek** [comment]
@evidolob, I fixed the test.

**Review by evidolob — APPROVED** [review]
